includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -121,6 +122,46 @@ public function init() {
 		}
 	}
 
+	/**
+	 * Updates the checkout order based on the request, to set the Apple Pay/Google Pay payment method title.
+	 *
+	 * @param \WC_Order        $order The order to be updated.
+	 * @param \WP_REST_Request $request Store API request to update the order.
+	 */
+	public function tokenized_cart_set_payment_method_type( \WC_Order $order, \WP_REST_Request $request ) {
+		if ( ! isset( $request['payment_method'] ) || 'woocommerce_payments' !== $request['payment_method'] ) {
+			return;
+		}
+
+		if ( empty( $request['payment_data'] ) ) {
+			return;
+		}
+
+		$payment_data = [];
+		foreach ( $request['payment_data'] as $data ) {
+			$payment_data[ sanitize_key( $data['key'] ) ] = wc_clean( $data['value'] );
+		}
+
+		if ( empty( $payment_data['payment_request_type'] ) ) {
+			return;
+		}
+
+		$payment_request_type = wc_clean( wp_unslash( $payment_data['payment_request_type'] ) );
+
+		$payment_method_titles = [
+			'apple_pay'  => 'Apple Pay',
+			'google_pay' => 'Google Pay',
+		];
+
+		$suffix = apply_filters( 'wcpay_payment_request_payment_method_title_suffix', 'WooPayments' );
+		if ( ! empty( $suffix ) ) {
+			$suffix = " ($suffix)";
+		}
+
+		$payment_method_title = isset( $payment_method_titles[ $payment_request_type ] ) ? $payment_method_titles[ $payment_request_type ] : 'Payment Request';
+		$order->set_payment_method_title( $payment_method_title . $suffix );
```

Comment:
This is working, so I presume we don't need to call `$order->save()` after this line. The payment method title property is correctly updated, but nonetheless, it might be safer to manually save the order anyways, since there are other hooks that fire when an order is saved that might be expected to act when an order's properties are updated. I'll leave it up to you, as it doesn't directly affect the functionality here and doesn't seem like not-saving could be too harmful in this instance...though, there may still be some subtle side-effects of not triggering the save function. 🤷

Reply:
Yeah, I didn't call `save()` on the order here because it is already called by the hook's caller: https://github.com/woocommerce/woocommerce/blob/9fb0aacc2518b6f5c3a2b4704ecdb438ab013346/plugins/woocommerce/src/StoreApi/Utilities/CheckoutTrait.php#L170